pkg/cache/snapshot_test.go

```diff
@@ -864,3 +864,447 @@ func TestSnapshotAddRemoveWorkload(t *testing.T) {
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 6_000},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 6_000,
+								},
+							},
+						},
+						"lend-b": {
+							Name:                          "lend-b",
+							Cohort:                        cohort,
+							Workloads:                     make(map[string]*workload.Info),
+							ResourceGroups:                cqCache.clusterQueues["lend-b"].ResourceGroups,
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 0},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 4_000,
+								},
+							},
+						},
+					},
+				}
+			}(),
+		},
+		"remove all then add workload, using quota over GuaranteedQuota": {
+			remove: []string{"/lend-a-1", "/lend-a-2", "/lend-a-3", "/lend-b-1"},
+			add:    []string{"/lend-a-2"},
+			want: func() Snapshot {
+				cohort := &Cohort{
+					Name:                          "lend",
+					AllocatableResourceGeneration: 2,
+					RequestableResources:          initialCohortResources,
+					Usage: FlavorResourceQuantities{
+						"default": {corev1.ResourceCPU: 3_000},
+					},
+				}
+				return Snapshot{
+					ClusterQueues: map[string]*ClusterQueue{
+						"lend-a": {
+							Name:                          "lend-a",
+							Cohort:                        cohort,
+							Workloads:                     make(map[string]*workload.Info),
+							ResourceGroups:                cqCache.clusterQueues["lend-a"].ResourceGroups,
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 9_000},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 6_000,
+								},
+							},
+						},
+						"lend-b": {
+							Name:                          "lend-b",
+							Cohort:                        cohort,
+							Workloads:                     make(map[string]*workload.Info),
+							ResourceGroups:                cqCache.clusterQueues["lend-b"].ResourceGroups,
+							FlavorFungibility:             defaultFlavorFungibility,
+							AllocatableResourceGeneration: 1,
+							Usage: FlavorResourceQuantities{
+								"default": {corev1.ResourceCPU: 0},
+							},
+							GuaranteedQuota: FlavorResourceQuantities{
+								"default": {
+									corev1.ResourceCPU: 4_000,
+								},
+							},
+						},
+					},
+				}
+			}(),
+		},
+	}
+	cmpOpts := append(snapCmpOpts,
+		cmpopts.IgnoreFields(ClusterQueue{}, "NamespaceSelector", "Preemption", "Status"),
+		cmpopts.IgnoreFields(Cohort{}),
```

Comment:
Something looks off here. Not sure what this is ignoring. I think we should ignore `ClusterQueues` within `Cohort` to avoid recursion.

Reply:
This is ignoring the whole `Cohort` struct. In this test function, all cqs belong to a same cohort, so there will be no difference. I think it's ok not to change this. WDYT?

Reply:
ping @alculquicondor

Reply:
Sounds resonable, modified.